ruma-api-macros/src/api/request/outgoing.rs

```diff
@@ -131,23 +131,36 @@ impl Request {
             .collect();
 
         for auth in &metadata.authentication {
-            if auth.value == "AccessToken" {
-                let attrs = &auth.attrs;
-                header_kvs.extend(quote! {
+            let attrs = &auth.attrs;
+            let auth_required = auth.value == "AccessToken";
+
+            let if_required_expr = if auth_required {
+                quote! { Some(access_token) }
+            } else {
+                quote! { None }
+            };
+
+            header_kvs.extend(quote! {{
+                let access_token = match access_token {
+                    #ruma_api::SendAccessToken::IfRequired(access_token) => #if_required_expr,
+                    #ruma_api::SendAccessToken::Always(access_token) => Some(access_token),
+                    #ruma_api::SendAccessToken::None => if #auth_required {
```

Comment:
Now you again introduced an if-else with a constant condition 😅

Reply:
Right :sweat_smile:

Reply:
Gave it another try :sweat_smile: